presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -133,16 +134,17 @@
     {
         BROADCAST,
         PARTITIONED,
-        AUTOMATIC;
+        AUTOMATIC,
+        AUTOMATIC_RESTRICTED;
```

Comment:
I don't think you need this knob anymore. Make `broadcastJoinMaxTableSize` nullable, and activate it when it is not null.

Reply:
It can't be nullable, but 0 disables the machanism